bar-trender-front-end/src/App.js

```diff
@@ -7,20 +7,44 @@ export default class App extends React.Component {
   state = {
     loading: true,
     qr: null,
+    error: null,
   };
   
+
   async componentDidMount()
   {
-    const url = 'http://localhost:8000/main/test/';
-    const response = await fetch(url);
-    const data = await response.blob();
-    this.setState({qr: URL.createObjectURL(data), loading: false})
-  }
 
+    var token = sessionStorage.getItem("token");
+    const id_establishment = this.props.idEstablishment;
+    const id_discount = this.props.idDiscount;
+    
+    if(!token){
+      this.setState({error: "Necesitas haber iniciado sesión para poder ver el descuento"})
+    }else{
+      const url = 'https://develop-backend-sprint-01.herokuapp.com/v1/establishments/'+id_establishment+'/discounts/'+id_discount+'/getQR';
+      const response = await fetch(url, {
+        method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+            'token': token,
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/BarTrender/BarTrenderFront/pullRequest?prid=7157793)
```suggestion
          "token": token,
```